### PR TITLE
Add gas fee stats

### PR DESCRIPTION
**Pull request by byterose**
PR for #15

### CODE REVIEW

**Comment by byterose** [comment]
Since retrieving data on-chain has proven to be too slow, we use the Etherscan API for now.

File: src/utils/index.ts [hunk 1]
```diff
@@ -52,10 +216,6 @@ export const decToBn = (dec: number, decimals = 18) => {
   return new BigNumber(dec).multipliedBy(new BigNumber(10).pow(decimals));
 };
 
-export const sleep = (ms: number) => {
-  return new Promise(resolve => setTimeout(resolve, ms));
-};
-
```
Comment (byterose) [inline src/utils/index.ts:58]: Moved sleep function 👆

File: src/views/Asset.vue [hunk 2]
```diff
@@ -26,6 +26,10 @@
 
       <Space size="md" />
 
+      <GasStats ref="gasStats" />
+
+      <Space size="md" />
```
Comment (byterose) [inline src/views/Asset.vue:31]: Size md looks better than 25 on mobile.

File: src/components/ugas/GasStats.vue [hunk 3]
```diff
@@ -0,0 +1,529 @@
...
+  border-radius: 9999px;
+  background-color: rgba(255, 255, 255, 1);
+  --toggle-icon-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06);
+  box-shadow: 0, 0, var(--toggle-icon-shadow);
+  box-shadow: 0 0;
+  transition-property: background-color, border-color, color, fill, stroke, opacity, box-shadow, transform;
+  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
+  transition-duration: 150ms;
+  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
+  transition-duration: 200ms;
```
Comment (ethedev) [inline src/components/ugas/GasStats.vue:414]:
```suggestion

```

File: src/components/ugas/GasStats.vue [hunk 4]
```diff
@@ -0,0 +1,529 @@
...
+          <svg style="height:1.25rem; width:1.25rem;" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
+            <path
+              stroke-linecap="round"
+              stroke-linejoin="round"
+              stroke-width="2"
+              d="M12 6V4m0 2a2 2 0 100 4m0-4a2 2 0 110 4m-6 8a2 2 0 100-4m0 4a2 2 0 110-4m0 4v2m0-6V4m6 6v10m6-2a2 2 0 100-4m0 4a2 2 0 110-4m0 4v2m0-6V4"
+            />
+          </svg>
+        </button>
+        <div v-show="showPopup" v-outside-click="{ exclude: ['stats-dropdown'], handler: 'onClose' }" class="dropdown-content">
```
Comment (ethedev) [inline src/components/ugas/GasStats.vue:74]: there is a scroll on the dropdown popup which doesnt need to be there unless the height was significantly smaller in size

**Comment by ethedev** [comment]
**Note:** on smaller devices, the stats menu button is too close to the warning text, converting it to non absolute position will solve this issue, and centering it to the middle of the screen will make it look better

File: src/components/ugas/GasStats.vue [hunk 5]
```diff
@@ -0,0 +1,518 @@
...
+          <div style="display:flex; flex-direction:row;">
+            <div style="padding-left:1rem; padding-right:1rem; padding-top:1rem; padding-bottom:1rem;">
+              <p class="dropdown-titel">
```
Comment (ethedev) [inline src/components/ugas/GasStats.vue:76]:
```suggestion
              <p class="dropdown-title">
```

File: src/components/ugas/GasStats.vue [hunk 6]
```diff
@@ -0,0 +1,518 @@
...
+  z-index: 2;
+  border-radius: 2px;
+  border: none;
+  right: 0;
+
+  @media (max-width: 800px) {
+    left: 0;
+  }
+}
+.dropdown-titel {
```
Comment (ethedev) [inline src/components/ugas/GasStats.vue:505]:
```suggestion
.dropdown-title {
```

File: src/components/ugas/GasStats.vue [hunk 7]
```diff
@@ -0,0 +1,518 @@
...
+.dropdown-content {
+  position: absolute;
+  background-color: #f9f9f9;
+  width: 225px;
+  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
+  background: var(--back-act);
+  z-index: 2;
+  border-radius: 2px;
+  border: none;
+  right: 0;
```
Comment (ethedev) [inline src/components/ugas/GasStats.vue:499]:
```suggestion
  position: absolute;
  width: 225px;
  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
  background: var(--back-act);
  z-index: 2;
  border-radius: 5px;
  border: none;
  right: 0;
```

File: src/components/ugas/GasStats.vue [hunk 8]
```diff
@@ -0,0 +1,518 @@
...
+  max-height: 25rem;
+  border-radius: 0.375rem;
+  padding-top: 0.25rem;
+  padding-bottom: 0.25rem;
+  font-size: 1rem;
+  line-height: 1.5rem;
+  overflow: auto;
+}
+.dropdown-list li:hover {
+  background-color: rgba(229, 231, 235, 1);
```
Comment (ethedev) [inline src/components/ugas/GasStats.vue:330]: thats for the moment, later these must be globally defined
```suggestion
  background-color: #f2eef0;
```

File: src/repo_palette/Space.vue [hunk 9]
```diff
@@ -23,6 +23,18 @@ export default {
...
+          return {
+            width: "32px",
+            height: "32px",
+          };
+        } else {
+          return {
+            width: "10px",
+            height: "10px",
+          };
+        }
```
Comment (ethedev) [inline src/repo_palette/Space.vue:37]: not a good practice also that isnt `dynamic` more like too specific. good way is to set 2 different classes on 2 different spaces handling it with mediaqueries, where one class will hide and other will show depending on the screen size

File: src/views/Asset.vue [hunk 10]
```diff
@@ -28,7 +28,8 @@
 
       <GasStats ref="gasStats" />
 
-      <Space size="dynamic" />
+      <Space class="showBigSpacer" size="md" />
+      <Space class="showSmallSpacer" size="10" />
```
Comment (ethedev) [inline src/views/Asset.vue:32]: 1. for now you can just name it `desktop-display` and `mobile-display` to be able to use it for any element
2. also non kamelcase for classes

File: src/views/Asset.vue [hunk 11]
```diff
@@ -1638,6 +1639,20 @@ export default {
 };
 </script>
 <style lang="scss" scoped>
+.showBigSpacer {
```
Comment (ethedev) [inline src/views/Asset.vue:1642]: for any reason for using `!important` here?

Comment (byterose) [reply src/views/Asset.vue:1642]: Yes, to overwrite the original Spacer styles.

**Pull request merged**